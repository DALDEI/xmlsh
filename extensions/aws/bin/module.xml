<module 
	module_version="1.0"
	name="aws"
	package="org.xmlsh.aws"
	require="2.0.0"
>
<classpath>
  <directory url="."/>
</classpath>
</module>

