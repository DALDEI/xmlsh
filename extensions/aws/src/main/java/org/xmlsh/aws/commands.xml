<?xml version="1.0"?>
<commands>
	<command name="s3ls" url="http://www.xmlsh.org/AWSs3ls">
		<synopsis>Lists files in a S3 bucket</synopsis>
		<usage>
			<para>s3ls [options] bucketname[/prefix]</para>
		</usage>
		<options>
			<option>
				<arg>-client client</arg>
				<para>Use an existing S3 client</para>
			</option>
			<option>
				<arg>-config file</arg>
				<para>read config information from file</para>
			</option>
			<option>
				<arg>-accessKey key</arg>
				<para>Specify the AWS Access Key</para>
			</option>
			<option>
				<arg>-secretKey key</arg>
				<para>Specify the AWS Secret Access Key</para>
			</option>
			<option>
				<arg>-r,-recurse</arg>
				<para>List all keys in the bucket (optionally starting with prefix)</para>
			</option>
			<option>
				<arg>-d,-delim delimiter</arg>
				<para>Specify a delimiter string (default "/")</para>
				<para>-r overrides this setting</para>
			</option>
			<option>
				<arg>bucketname</arg>
				<para>Name of the AWS bucket to list</para>
			</option>


		</options>
	</command>


</commands>