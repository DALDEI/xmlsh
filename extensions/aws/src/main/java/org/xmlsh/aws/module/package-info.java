/**
 * 
 */
/**
 * @author DLEE
 * 
 */
@Module(name="aws",moduleClass="AWSModule")
package org.xmlsh.aws.module;
import org.xmlsh.annotations.Module;

