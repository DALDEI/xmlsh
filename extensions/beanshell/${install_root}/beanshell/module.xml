<module 
	module_version="1.0"
	name="beanshell"
	require="2.0.0"
>
<classpath>
  <directory url="."/>
</classpath>
<packages>
  <package>org.xmlsh.bsh</package>
 </packages>
<main>
  <classname>org.xmlsh.bsh.module.BSHModule</classname>
</main>
</module>

