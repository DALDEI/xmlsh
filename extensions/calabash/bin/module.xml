<module 
	module_version="1"
	name="calabash"
	package="org.xmlsh.calabash"
	require="1.2.1"
>
<classpath>
  <file url="calabash_ext.jar"/>
  <file url="calabash.jar"/>
  <file url="commons-httpclient-3.1.jar"/>
  <file url="commons-logging-1.1.1.jar"/>
  <file url="commons-logging-api-1.1.1.jar"/>
  <file url="commons-codec-1.3.jar"/>
  <file url="isorelax.jar"/>
  <file url="msv.jar"/>
  <file url="tagsoup-1.2.jar"/>
  <file url="xep.jar"/>
  
            
</classpath>
</module>

