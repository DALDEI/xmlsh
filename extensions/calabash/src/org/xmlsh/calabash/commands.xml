<?xml version="1.0"?>
<commands>
	<command name="xproc" url="http://www.xmlsh.org/CalabashXproc">
		<synopsis>Executes an xproc pipeline</synopsis>
		<usage>
			<para>xproc [options] pipeline</para>
		</usage>
		<options>
			<option>
				<arg>-b,-base uri</arg>
				<para>Override the base-uri of the xproc pipeline file (to simulate executing a remote pipeline from a local file)</para>
			</option>
			<option>
				<arg>-n</arg>
				<para>No input.  Do not read any input ports for pipeline input</para>
			</option>
			<option>
				<arg>-iw,-iwrap</arg>
				<para>Expect input documents to be wrapped by a root node.  This allows a sequence of documents on an input port.</para>
			</option>
			<option>
				<arg>-ow,-owrap</arg>
				<para>Wrap output documents with a root node "wrap".  This allows a sequence of documents on an output port</para>
			</option>

			<option>
				<arg>-o,-option name=value</arg>
				<para>Pass an option to the pipeline.  May occur multiple times</para>
			</option>


			<option>
				<arg>pipeline</arg>
				<para>path to xproc pipeline file</para>
			</option>

		</options>
	</command>
	
</commands>