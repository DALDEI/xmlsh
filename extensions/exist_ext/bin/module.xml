<module 
	module_version="0.7"
	name="exist"
	package="org.xmlsh.exist" require="1.2.3">
<classpath>
  <file url="exist_ext.jar"/>
</classpath>
</module>

