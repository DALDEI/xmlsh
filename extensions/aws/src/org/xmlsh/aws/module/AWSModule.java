package org.xmlsh.aws.module;

import java.util.List;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.xmlsh.annotations.Function;
import org.xmlsh.core.AbstractBuiltinFunction;
import org.xmlsh.core.CoreException;
import org.xmlsh.core.XClassLoader;
import org.xmlsh.core.XValue;
import org.xmlsh.sh.module.ExternalModule;
import org.xmlsh.sh.module.ModuleConfig;
import org.xmlsh.sh.shell.Shell;

@org.xmlsh.annotations.Module
public class AWSModule extends ExternalModule {

     static Logger mLogger = LogManager.getLogger();
	
	public AWSModule(ModuleConfig config, XClassLoader loader) throws CoreException {
		super(config, loader);
	  mLogger.entry(config, loader);
	
		
	}

	@Override
	public void onInit(Shell shell, List<XValue> args) throws Exception {
		super.onInit(shell, args);
	  mLogger.entry(shell, args);
	}

	@Override
	public void onLoad(Shell shell) {
		super.onLoad(shell);
        mLogger.entry(shell);


	}

	@Function( name="ec2-client")
	public static class ec2Client extends AbstractBuiltinFunction {
		  @Override
		  public XValue run(Shell shell, List<XValue> args) throws Exception
		  {

			  return null ;
			  
		  }
		
	
	}
		


}
