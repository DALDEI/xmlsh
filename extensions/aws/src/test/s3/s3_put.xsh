# Test of s3 CreateBucket

import module aws=aws
import commands posix

#foo has name of bucket just created
bucket=xmlsh-org-test-$RANDOM
aws:s3-create-bucket $bucket
foo=s3://$bucket

rm -rf $TMPDIR/awss3
mkdir $TMPDIR/awss3
cd $TMPDIR/awss3

# make some files

cat > f1 <<EOF
This is f1
EOF

cat > f2 <<EOF
This is f2
EOF

cat > f3 <<EOF
This is f3
EOF

# step 1 - verify single file upload

aws:s3put f1 $foo
aws:s3get $foo/f1 f1back

xcmp f1 f1back && echo Step 1 Success compared Success

aws:s3Delete $foo/f1

rm -f f1back

# step 2 - verify multiple single file upload

aws:s3put f1 f2 $foo
aws:s3get $foo/f1 f1back
aws:s3get $foo/f2 f2back

xcmp f1 f1back && echo Step 2-f1 Success compared Success
xcmp f2 f2back && echo Step 2-f2 Success compared Success

aws:s3-delete $foo/f1
aws:s3-delete $foo/f2
rm -f f1back
rm -f f2back

# step 3 - verify multifile directory upload

cd ..

aws:s3put -r awss3 $foo

cd awss3

aws:s3get $foo/awss3/f1 f1back
aws:s3get $foo/awss3/f2 f2back
aws:s3get $foo/awss3/f3 f3back

xcmp f1 f1back && echo Step 3-f1 Success compared Success
xcmp f2 f2back && echo Step 3-f2 Success compared Success
xcmp f3 f3back && echo Step 3-f3 Success compared Success

aws:s3-delete -r $foo/awss3

aws:s3-delete-bucket $bucket
rm -rf $TMPDIR/awss3

exit 0