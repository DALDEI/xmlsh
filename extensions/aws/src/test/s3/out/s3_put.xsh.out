Step 1 Success compared Success
Step 2-f1 Success compared Success
Step 2-f2 Success compared Success
Step 3-f1 Success compared Success
Step 3-f2 Success compared Success
Step 3-f3 Success compared Success
